go_utils/photo_download.py

```diff
@@ -286,15 +366,39 @@ def get_lc_download_targets(
     targets = set()
 
     def get_photo_args(url, latitude, longitude, direction, date, lc_id):
-        if pd.isna(url) or "https" not in url:
-            return
-
-        date_str = pd.to_datetime(str(date)).strftime("%Y-%m-%d")
-        photo_id = get_globe_photo_id(url)
+        if not pd.isna(url) and "https" in url:
+            date_str = pd.to_datetime(str(date)).strftime("%Y-%m-%d")
+            photo_id = get_globe_photo_id(url)
 
-        name = f"lc_{direction}_{round(latitude, 5)}_{round(longitude, 5)}_{date_str}_{lc_id}_{photo_id}.png"
-        name = remove_bad_characters(name)
-        targets.add((url, directory, name))
+            # f"lc_{direction}_{round(latitude, 5)}_{round(longitude, 5)}_{date_str}_{lc_id}_{photo_id}.png"
+            def build_name():
+                name = "lc_"
+                if additional_name_stem and additional_name_stem != "":
+                    name += f"{additional_name_stem}_"
+
+                if include_in_name:
+                    for field in list(include_in_name):
+                        if field in list(name_fields):
+                            name += f"{name_fields[field]}_"
+                name += f"{photo_id}.png"
+                name = remove_bad_characters(name)
+                return name
+
+            name_fields = {
+                "direction": direction,
+                "latitude": round(latitude, 5),
+                "longitude": round(longitude, 5),
+                "date_str": date_str,
+                "lc_id": lc_id,
+            }
+
+            if not pd.isna(photo_id) and int(photo_id) >= 0:
+                name = build_name()
+                targets.add((url, directory, name))
+            else:
+                print("Invalid photoID: " + photo_id)
+        else:
+            print("Invalid URL: " + str(url))
```

Comment:
See warning comment.

Reply:
see https://github.com/IGES-Geospatial/globe-observer-utils/pull/73#discussion_r776763168

Reply:
for this warning -- rather than warning for each invalid photo, I provided a warning at the end of the iteration reporting the number of invalid/skipped photos -- This way there is less console output for the User